test/python/circuit/test_uc.py

```diff
@@ -85,6 +86,21 @@ def test_global_phase_ucg(self):
 
         self.assertTrue(np.allclose(unitary_desired, unitary))
 
+    def test_inverse_ucg(self):
```

Comment:
Could you also add a test for the isometry?

Reply:
There are already tests for the isometries, but they are ignoring the phase because of the problem fixed here. I will change those tests to consider the phase.